src/game_api/level_api.cpp

```diff
@@ -105,13 +112,13 @@ std::array g_community_tile_codes{
     CommunityTileCode{"hundun", "ENT_TYPE_MONS_HUNDUN"},
     CommunityTileCode{"scarab", "ENT_TYPE_MONS_SCARAB"},
     CommunityTileCode{"cosmic_jelly", "ENT_TYPE_MONS_MEGAJELLYFISH"},
-    CommunityTileCode{"ghost", "ENT_TYPE_MONS_GHOST"},
-    CommunityTileCode{"ghost_med_sad", "ENT_TYPE_MONS_GHOST_MEDIUM_SAD"},
-    CommunityTileCode{"ghost_med_happy", "ENT_TYPE_MONS_GHOST_MEDIUM_HAPPY"},
-    CommunityTileCode{"ghost_small_angry", "ENT_TYPE_MONS_GHOST_SMALL_ANGRY"},
-    CommunityTileCode{"ghost_small_sad", "ENT_TYPE_MONS_GHOST_SMALL_SAD"},
-    CommunityTileCode{"ghost_small_surprised", "ENT_TYPE_MONS_GHOST_SMALL_SURPRISED"},
-    CommunityTileCode{"ghost_small_happy", "ENT_TYPE_MONS_GHOST_SMALL_HAPPY"},
+    CommunityTileCode{"ghost", "ENT_TYPE_MONS_GHOST", g_spawn_ghost<GHOST_BEHAVIOR::SAD>},
+    CommunityTileCode{"ghost_med_sad", "ENT_TYPE_MONS_GHOST_MEDIUM_SAD", g_spawn_ghost<GHOST_BEHAVIOR::SAD>},
+    CommunityTileCode{"ghost_med_happy", "ENT_TYPE_MONS_GHOST_MEDIUM_HAPPY", g_spawn_ghost<GHOST_BEHAVIOR::HAPPY>},
```

Comment:
I think you have to look at it as three different sets of states (ghost:0 ; medium: 0,1; small: 0, 1, 2, 3)
The original comment named the states in context of the small ghost.

Reply:
I see. That's still really weird since I'd expect that all ghosts use the same statemachine but now that implies that the state machine code either is separate or that it has to take into account the type id. Either way surprising to me. I'll fix it later